### ISSUE TITLE
[Gitlab] Error while decorating the request 

### ISSUE THREAD

**Issue by sfaqer** [mc1arke/sonarqube-community-branch-plugin#445]
**Describe the bug**
Found error in CE logs, MR not decorated

**Expected behavior**
MR successful decorated

**Software Versions**
 - SonarQube Version: 9.0.1 (build 46107)
 - Plugin Version: 1.9.0
 - Docker image: mc1arke/sonarqube-with-community-branch-plugin:9.0-community

**Additional context**
```
sonarqube_1  | 2021.09.15 08:05:11 INFO  ce[AXvofOU_UQuhcpSys54E][c.g.m.s.p.c.p.PullRequestPostAnalysisTask] using pull request decorator com.github.mc1arke.sonarqube.plugin.ce.pullrequest.gitlab.GitlabMergeRequestDecorator
sonarqube_1  | 2021.09.15 08:05:12 ERROR ce[AXvofOU_UQuhcpSys54E][c.g.m.s.p.a.g.GitlabRestClient] Gitlab response status did not match expected value. Expected: 201\nHttpResponseProxy{HTTP/1.1 400 Bad Request [Server: nginx, Date: Wed, 15 Sep 2021 08:05:12 GMT, Content-Type: application/json, Content-Length: 107, Connection: keep-alive, Cache-Control: no-cache, Vary: Origin, X-Content-Type-Options: nosniff, X-Frame-Options: SAMEORIGIN, X-Request-Id: 01FFM7T0RH6WGZS0KPXTMBQ44E, X-Runtime: 0.074482] ResponseEntityProxy{[Content-Type: application/json,Content-Length: 107,Chunked: false]}}\n{"message":"400 Bad request - Note {:line_code=\u003e[\"can't be blank\", \"must be a valid line code\"]}"}
sonarqube_1  | 2021.09.15 08:05:12 ERROR ce[AXvofOU_UQuhcpSys54E][o.s.c.t.p.a.p.PostProjectAnalysisTasksExecutor] Execution of task class com.github.mc1arke.sonarqube.plugin.ce.pullrequest.PullRequestPostAnalysisTask failed
sonarqube_1  | java.lang.IllegalStateException: An unexpected response code was returned from the Gitlab API - Expected: 201, Got: 400
sonarqube_1  |  at com.github.mc1arke.sonarqube.plugin.almclient.gitlab.GitlabRestClient.validateResponse(GitlabRestClient.java:230)
sonarqube_1  |  at com.github.mc1arke.sonarqube.plugin.almclient.gitlab.GitlabRestClient.lambda$addMergeRequestDiscussion$0(GitlabRestClient.java:115)
sonarqube_1  |  at com.github.mc1arke.sonarqube.plugin.almclient.gitlab.GitlabRestClient.entity(GitlabRestClient.java:175)
sonarqube_1  |  at com.github.mc1arke.sonarqube.plugin.almclient.gitlab.GitlabRestClient.addMergeRequestDiscussion(GitlabRestClient.java:115)
sonarqube_1  |  at com.github.mc1arke.sonarqube.plugin.ce.pullrequest.gitlab.GitlabMergeRequestDecorator.submitCommitNoteForIssue(GitlabMergeRequestDecorator.java:155)
sonarqube_1  |  at com.github.mc1arke.sonarqube.plugin.ce.pullrequest.gitlab.GitlabMergeRequestDecorator.submitCommitNoteForIssue(GitlabMergeRequestDecorator.java:52)
sonarqube_1  |  at com.github.mc1arke.sonarqube.plugin.ce.pullrequest.DiscussionAwarePullRequestDecorator.lambda$decorateQualityGateStatus$5(DiscussionAwarePullRequestDecorator.java:99)
sonarqube_1  |  at java.base/java.util.ArrayList.forEach(Unknown Source)
sonarqube_1  |  at com.github.mc1arke.sonarqube.plugin.ce.pullrequest.DiscussionAwarePullRequestDecorator.decorateQualityGateStatus(DiscussionAwarePullRequestDecorator.java:99)
sonarqube_1  |  at com.github.mc1arke.sonarqube.plugin.ce.pullrequest.PullRequestPostAnalysisTask.finished(PullRequestPostAnalysisTask.java:160)
sonarqube_1  |  at org.sonar.ce.task.projectanalysis.api.posttask.PostProjectAnalysisTasksExecutor.executeTask(PostProjectAnalysisTasksExecutor.java:110)
sonarqube_1  |  at org.sonar.ce.task.projectanalysis.api.posttask.PostProjectAnalysisTasksExecutor.finished(PostProjectAnalysisTasksExecutor.java:101)
sonarqube_1  |  at org.sonar.ce.task.step.ComputationStepExecutor.executeListener(ComputationStepExecutor.java:91)
sonarqube_1  |  at org.sonar.ce.task.step.ComputationStepExecutor.execute(ComputationStepExecutor.java:63)
sonarqube_1  |  at org.sonar.ce.task.projectanalysis.taskprocessor.ReportTaskProcessor.process(ReportTaskProcessor.java:81)
sonarqube_1  |  at org.sonar.ce.taskprocessor.CeWorkerImpl$ExecuteTask.executeTask(CeWorkerImpl.java:212)
sonarqube_1  |  at org.sonar.ce.taskprocessor.CeWorkerImpl$ExecuteTask.run(CeWorkerImpl.java:194)
sonarqube_1  |  at org.sonar.ce.taskprocessor.CeWorkerImpl.findAndProcessTask(CeWorkerImpl.java:160)
sonarqube_1  |  at org.sonar.ce.taskprocessor.CeWorkerImpl$TrackRunningState.get(CeWorkerImpl.java:135)
sonarqube_1  |  at org.sonar.ce.taskprocessor.CeWorkerImpl.call(CeWorkerImpl.java:87)
sonarqube_1  |  at org.sonar.ce.taskprocessor.CeWorkerImpl.call(CeWorkerImpl.java:53)
sonarqube_1  |  at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:125)
sonarqube_1  |  at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:69)
sonarqube_1  |  at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:78)
sonarqube_1  |  at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
sonarqube_1  |  at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
sonarqube_1  |  at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source)
sonarqube_1  |  at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
sonarqube_1  |  at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
sonarqube_1  |  at java.base/java.lang.Thread.run(Unknown Source)
sonarqube_1  | 2021.09.15 08:05:12 INFO  ce[AXvofOU_UQuhcpSys54E][o.s.c.t.p.a.p.PostProjectAnalysisTasksExecutor] Pull Request Decoration | status=FAILED | time=306ms
```

**Comment by mc1arke** [comment]
I'd need some more details to help replicate this:
1. Does this consistently fail for this MR if you were to re-run it?
2. Do you have this fail for other MRs?
3. Are you doing anything like amending commits whilst your build is running?

**Comment by sfaqer** [comment]
> Does this consistently fail for this MR if you were to re-run it?

Yes

> Do you have this fail for other MRs?

Yes

> Are you doing anything like amending commits whilst your build is running?

no

btw i found error and fix it in #446 =)

**Comment by mc1arke** [comment]
Released in 1.10.0 of the plugin
